contracts/helpers/MultiBlockHarvester.sol

```diff
@@ -0,0 +1,232 @@
+                                                        GOVERNOR FUNCTIONS
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Set the maximum amount of stablecoins that can be minted in a single transaction
+     * @param newMaxMintAmount new maximum amount of stablecoins that can be minted in a single transaction
+     */
+    function setMaxMintAmount(uint256 newMaxMintAmount) external onlyGovernor {
+        maxMintAmount = newMaxMintAmount;
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+                                                        GUARDIAN FUNCTIONS
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Set the deposit address for a yieldBearingAsset
+     * @param yieldBearingAsset address of the yieldBearingAsset
+     * @param newDepositAddress address to deposit to receive yieldBearingAsset
+     */
+    function setYieldBearingToDepositAddress(
+        address yieldBearingAsset,
+        address newDepositAddress
+    ) external onlyGuardian {
+        yieldBearingToDepositAddress[yieldBearingAsset] = newDepositAddress;
+    }
+
+    /**
+     * @notice Toggle the trusted status of an address
+     * @param trusted address to toggle the trusted status
+     */
+    function toggleTrusted(address trusted) external onlyGuardian {
+        isTrusted[trusted] = !isTrusted[trusted];
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+                                                        TRUSTED FUNCTIONS
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Initiate a rebalance
+     * @param scale scale to apply to the rebalance amount
+     * @param yieldBearingAsset address of the yieldBearingAsset
+     */
+    function harvest(address yieldBearingAsset, uint256 scale, bytes calldata) external onlyTrusted {
+        if (scale > 1e9) revert InvalidParam();
+        YieldBearingParams memory yieldBearingInfo = yieldBearingData[yieldBearingAsset];
+        (uint8 increase, uint256 amount) = _computeRebalanceAmount(yieldBearingAsset, yieldBearingInfo);
+        amount = (amount * scale) / 1e9;
+        if (amount == 0) revert ZeroAmount();
+
+        try transmuter.updateOracle(yieldBearingAsset) {} catch {}
+        _rebalance(increase, yieldBearingAsset, yieldBearingInfo, amount);
+    }
+
+    /**
+     * @notice Finalize a rebalance
+     * @param yieldBearingAsset address of the yieldBearingAsset
+     */
+    function finalizeRebalance(address yieldBearingAsset, uint256 balance) external onlyTrusted {
```

Comment:
There should be some storage here to be sure we finalise rebalance toward an asset that we start with
Something in the rebalance that maps asset to what amount max can be rebalanced

Reply:
Ah no this is ok in fact